Create Modal integration tile (#2135)
DataDog/integrations-extras: modal/README.md

# Modal

## Overview

[Modal][1] lets you run generative AI models, large-scale batch jobs, job

> queues, and much more. The Datadog integration collects logs and metrics from

Reviewer: ```suggestion
queues, and much more. This integration collects logs and metrics from
```

> your applications for you to observe directly in your Datadog account.

Reviewer: Can you add a bit more context here about what data is  being sent in/what it represents, and why it's valuable for customers?

## Setup

### Installation

To set up the Modal integration:

> 1. On the [Datadog Integrations page][2], search for the **Modal** tile and

Reviewer: ```suggestion
1. Navigate to the [**Modal** tile][2] on the Datadog Integrations page and
```

   click **Install Integration**.

2. Click **Connect Accounts** to begin authorization of this integration. You
   will be redirected to log into [Modal][1], and once logged in, you'll be
   redirected to the Datadog authorization page.

3. Click the **Authorize** button to complete setup.

Logs and metrics from your Modal apps should now start appearing in Datadog.

## Data Collected

### Metrics

Modal collects utilization metrics for applications/functions that run on Modal.

> You can view these metrics on your [Modal][1] dashboard.

Reviewer: It may be more helpful to mention where users can find these metrics and other data within Datadog, rather than within Modal.

### Logs

Modal collects audit logs and application logs, both of which you can access through your [Modal][1] dashboard. Audit logs can also be accessed in Datadog.

### Events

Modal does not include any events.

## Uninstallation

Once this integration has been uninstalled, any previous authorizations are
revoked and logs/metrics stop being emitted to Datadog.

1. On the **Configure** tab in the **Modal** integration tile in Datadog, click **Uninstall Integration**.

2. Ensure that all API keys associated with this integration have been disabled
   by searching for the integration name on the [API Keys page][4].

## Troubleshooting

Need help? Contact [Modal support][3].

[1]: https://modal.com

> [2]: https://app.datadoghq.com/integrations

Reviewer: ```suggestion
[2]: https://app.datadoghq.com/integrations?integrationId=modal
```

# Modal

## Overview

[Modal][1] lets you run generative AI models, large-scale batch jobs, job
queues, and much more. This integration collects metrics (CPU, memory
or GPU use for example) and logs (stdout/stderr logging, audit logs) from your

> applications for you to observe directly in your Datadog account.

Reviewer 2: The overview should include: 

1. What does your product do (in 1-2 sentences)? This has been done. 
2. What will customers get from this integration, and why is it valuable to them? This should be clear to new customers who are looking at the tile. 
3. What types of data will your integration send in or pull out, **and what's the value of that data?**

Author: Appended a few sentences and re-edited the overview text. It's very general sounding given "modal applications" are effectively any python program, so for instance if a program logged "<N> tasks completed" and then set up a datadog filter for N and graphed that overtime, that's valuable to the author of the application. It's specific to Modal in that we're running the code for you, and emitting those logs to your datadog account, but it's valuable in anyway you choose to use it.

## Setup

### Installation

To set up the Modal integration:

1. Navigate to the [**Modal** tile][2] on the Datadog Integrations page and
   click **Install Integration**.

2. Click **Connect Accounts** to begin authorization of this integration. You
   will be redirected to log into [Modal][1], and once logged in, you'll be
   redirected to the Datadog authorization page.

3. Click the **Authorize** button to complete setup.

Logs and metrics from your Modal apps should now start appearing in Datadog.

## Data Collected

### Metrics

Modal collects GPU, CPU, and memory utilization metrics for
applications/functions that run on Modal.

### Logs

> Modal collects audit logs and all application logs (all stdout/stderr logging).

Reviewer 2: To confirm, the integration is sending logs to Datadog?

Author: Yes.

# Modal

## Overview

> [Modal][1] lets you run generative AI models, large-scale batch jobs, job

Reviewer: Might be helpful to be specific here about how Modal runs the infrastructure to power these jobs

Author: It's feels slightly wordy to be specific here - we make use our custom built containerization tech and filesystems to do so - details we generally don't make apparent to users, hoping it's all seamless. Given we've linked to the modal homepage (where the docs talk through some of these details), what do you think leaving it as is?

Reviewer: Gotcha - definitely dont have to do go into details about the custom containerization, but I think it would add context to the idea that "Modal lets you run xyz" just to explain how it does that. Fine leaving as-is if you think this is descriptive enough!

Author: Ack, I'll leave it as is, but mostly because I'm finding it hard to phrase it too differently without losing folks.

> queues, and much more. This integration collects metrics (CPU, memory
> or GPU use for example) and logs (stdout/stderr logging from your modal
> applications, or audit logs for your account), for you to observe the data
> directly in your Datadog account. This lets you visualize through Datadog
> dashboards metrics about your modal applications, and also set up alerts.

Reviewer: ```suggestion
queues, and much more. This integration collects metrics (CPU, memory
or GPU use for example) and logs (stdout/stderr logging from your modal
applications, or audit logs for your account) that you can visualize through Datadog
dashboards and set up alerts for with Datadog monitors.
```

your applications for you to observe directly in your Datadog account.

## Setup

### Installation

To set up the Modal integration:

1. Go to your [Datadog Integrations page][2], search for the **Modal** tile and
   click **Install Integration**.

2. Click **Connect Accounts** to begin authorization of this integration. You
   will be redirected to log into [Modal][1], and once logged in, you'll be
   redirected to the Datadog authorization page.

3. Click the **Authorize** button to complete setup.

Logs and metrics from your Modal apps should now start appearing in Datadog.

## Data Collected

### Metrics

Modal collects utilization metrics for applications/functions run on Modal --
you can currently view them on your [Modal][1] dashboard directly. We're
working on surfacing this to Datadog shortly.

### Logs

Modal collects both audit logs and application logs, both accessble through
your [Modal][1] dashboard. Audit logs are currently accessible through Datadog
as well. We're working on surfacing application logs through shortly.

### Events

Modal does not include any events.

## Uninstallation

Once this integration has been uninstalled, any previous authorizations are

> revoked and logs/metrics are 

Author: Oops, done.
